### ISSUE TITLE
Dividing Rationals produces error related to implementation details

### ISSUE THREAD

**Issue by mbarbar** [JuliaLang/julia#51730]
In trying to perform
```
Rational{Int8}(-128) / Rational{Int8}(0)
```
we get an `OverflowError`
```
ERROR: OverflowError: checked arithmetic: cannot compute |x| for x = -128::Int8
Stacktrace:
 [1] checked_abs
   @ ./checked.jl:128 [inlined]
 [2] gcd(a::Int8, b::Int8)
   @ Base ./intfuncs.jl:52
 [3] divgcd
   @ ./rational.jl:44 [inlined]
 [4] //(x::Rational{Int8}, y::Rational{Int8})
   @ Base ./rational.jl:73
 [5] /(x::Rational{Int8}, y::Rational{Int8})
   @ Base ./rational.jl:360
 [6] top-level scope
   @ REPL[2]:1
```
within the implementation of division, instead of the expected `DivideError`. `checked_abs` is obviously not part of the division contract.

Julia installed from the dmg, versioninfo:
```
julia> versioninfo()
Julia Version 1.9.2
Commit e4ee485e909 (2023-07-05 09:39 UTC)
Platform Info:
  OS: macOS (arm64-apple-darwin22.4.0)
  CPU: 12 × Apple M2 Max
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-14.0.6 (ORCJIT, apple-m1)
  Threads: 1 on 8 virtual cores
```

**Comment by nhz2** [comment]
I think the expected result is `-1//0`

**Comment by nhz2** [comment]
This can be closed now, it was fixed by #51944
